scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -869,6 +932,12 @@ private void processRecord(Record record, Connection connection) {
 			LOGGER.info("Unexpected error occurred while processing record from peer [{}]",
 					record.getPeerAddress(), e);
 			terminateConnection(connection, e, AlertLevel.FATAL, AlertDescription.INTERNAL_ERROR);
+		} catch (GeneralSecurityException e) {
+			LOGGER.info("error occurred while processing record from peer [{}]",
+					record.getPeerAddress(), e);
```

Comment:
That's the receiving part, so I would not terminate the connection, because this may by used as attack vector. That's somehow the opposite of the catch RuntimeException in receiveNextDatagramFromNetwork. But this PR is more about introducing the cid, than the cleanup of exception handling. I tried to keep it as I found it, may be it gets cleaned up later in an other PR, if we find that it must be improved.